internal/storage/v2/elasticsearch/tracestore/from_dbmodel.go

```diff
@@ -58,21 +55,22 @@ func dbProcessToResource(process dbmodel.Process, resource pcommon.Resource) {
 	dbTagsToAttributes(tags, attrs)
 }
 
-func dbSpansToSpans(dbSpans []*dbmodel.Span, resourceSpans ptrace.ResourceSpansSlice) error {
-	for _, dbSpan := range dbSpans {
+func dbSpansToSpans(dbSpans []dbmodel.Span, resourceSpans ptrace.ResourceSpansSlice) error {
+	for i := range dbSpans {
 		// TODO why is this needed?
-		if dbSpan == nil || reflect.DeepEqual(dbSpan, blankJaegerProtoSpan) {
+		if reflect.DeepEqual(dbSpans[i], dbmodel.Span{}) {
```

Comment:
let's just delete this, it doesn't make sense for data from DB to come as empty dbSpan